File: frontend/flight-booking/src/model/BookingDTO.ts
import { CustomerDTO } from "./CustomerDTO";
import { FlightScheduleDTO } from "./FlightScheduleDTO";

export interface BookingDTO {
    id?: number;
    bookingReference: string;
    bookingDate: Date;
    totalPrice: number;
    bookingStatus: string;
    flightScheduleDTO?: FlightScheduleDTO;
    passenger: CustomerDTO;



}

File: backend/api/src/main/java/edu/miu/cs425/flightbookingsystem/dto/BookingDTO.java
package edu.miu.cs425.flightbookingsystem.dto;

import edu.miu.cs425.flightbookingsystem.model.BookingStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record BookingDTO(
        Long id,
        String bookingReference,
        LocalDateTime bookingDate,
        BigDecimal totalPrice,
        BookingStatus bookingStatus,
        CustomerDTO passenger,
        FlightScheduleDTO flightScheduleDTO
) {
}
